packages/inventory-next/src/models/inventory-item.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  BeforeCreate,
+  Collection,
+  Entity,
+  Filter,
+  OnInit,
+  OneToMany,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
+
+import { DAL } from "@medusajs/types"
+import { InventoryLevel } from "./inventory-level"
+
+const InventoryItemDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: "inventory_item",
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+
+const InventoryItemSkuIndex = createPsqlIndexStatementHelper({
+  tableName: "inventory_item",
+  columns: "sku",
+  unique: true,
+})
+
+type InventoryItemOptionalProps = DAL.SoftDeletableEntityDateColumns
+
+@Entity()
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export class InventoryItem {
+  [OptionalProps]: InventoryItemOptionalProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({
+    onCreate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  created_at: Date
+
+  @Property({
+    onCreate: () => new Date(),
+    onUpdate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  updated_at: Date
+
+  @InventoryItemDeletedAtIndex.MikroORMIndex()
+  @Property({ columnType: "timestamptz", nullable: true })
+  deleted_at: Date | null = null
+
+  @InventoryItemSkuIndex.MikroORMIndex()
+  @Property({ columnType: "text", nullable: true })
```

Comment:
A very general question: We have a lot of these fields on the product model as well, which causes for a lot of duplication and chance of these two getting out of sync. Which one is the source of truth? I don't know much about the logic behind the inventory module, but I see the benefit of having this information on the inventory item itself, however we have to ensure the two are in sync.

Or should the inventory item just reference the product?

Reply:
there is a relation between product-variants and inventory-items `productVariantInventoryItem` (described in the link-modules package`). Given that both packages are installed and running this should be the source of truth rather than then product-variant fields. 

If we expect to always be running both packages in v2 it might make sense to remove the fields from the product-module @adrien2p ?

Reply:
Yeah the idea was mostly to be backward compatible since the product module was also used in v1, but more generally the inventory on module should probably not even exists and creates some nasty side effects when customers are buying the same variants. Probably in v2 we might have either inventory management on or off, maybe we should validate that more globally @olivermrbl

Reply:
A product variant can be made up from x number of inventory items. An inventory item is the physical thing being shipped and a product variant is the commercial representation of that thing. For example, a customer will buy an "oval table" which is the product variant, but the variant consists of 4 x "table legs" and 1 x "table top". 

The duplication of these fields is, as Adrien mentions, because we introduced the inventory module as a way to remain backward compatible with what was there in the past (where inventory could also be tracked on the `inventory_quantity` field on product variants). In v2, we can enforce that anyone who needs inventory tracking should use the inventory module, so cleaning up these fields would be sensible.

I would keep SKU on product variants since that could still make sense but, hs_code, origin_country, measurements, etc. would all make sense to remove.